dotnet/targets/Xamarin.Shared.Sdk.targets

```diff
@@ -2558,6 +2559,33 @@
 		</PropertyGroup>
 	</Target>
 
+	<!--
+	***********************************************************************************************
+	ComputeAvailableDevices
+
+	Target that queries available devices and simulators.
+	This target is called by 'dotnet run' to support device selection.
+	Returns @(Devices) items with metadata: ???????
+
+	See: https://github.com/dotnet/sdk/blob/2b9fc02a265c735f2132e4e3626e94962e48bdf5/documentation/specs/dotnet-run-for-maui.md
+	***********************************************************************************************
+	-->
+	<Target Name="ComputeAvailableDevices"
+		DependsOnTargets="_DetectSdkLocations;_GenerateBundleName;"
+		Returns="@(Devices)">
+		<GetAvailableDevices
+			Condition="'$(IsMacEnabled)' == 'true'"
+			SessionId="$(BuildSessionId)"
```

Comment:
That is a question... Will this work on Windows? Ideally, it could, but that could be future PR.